comp/process/agent/flare.go

```diff
@@ -0,0 +1,55 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//nolint:revive
+package agent
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	flaretypes "github.com/DataDog/datadog-agent/comp/core/flare/types"
+	"github.com/DataDog/datadog-agent/pkg/process/checks"
+)
+
+// FlareHelper
+type FlareHelper struct {
+	m      sync.Mutex
+	Checks []checks.Check
+}
+
+// NewNewFlareHelper
+func NewFlareHelper(checks []checks.Check) *FlareHelper {
+	return &FlareHelper{Checks: checks}
+}
+
+// FillFlare is the callback function for the flare.
+func (fh *FlareHelper) FillFlare(fb flaretypes.FlareBuilder) error {
+	fh.m.Lock()
+	defer fh.m.Unlock()
+
+	for _, check := range fh.Checks {
```

Comment:
I assume this doesn't take too much time. Is there any concern with blocking this callback with a lock?

Reply:
I added this because it seemed to be a general practice (not always) and the process agent does alter the same files when the flare is made. However, it should be impossible for it to happen concurrently due to the change I made to `archive.go`. I think this would be safe to remove.